sites/public/src/pages/listings.tsx

```diff
@@ -72,6 +77,10 @@ export default function ListingsPage(props: ListingsProps) {
 }
 
 export async function getServerSideProps() {
+  // Hack alert: fetchOpenListings and fetchClosedListings call these two
+  // async functions concurrently, which causes a race condition, so
+  // calling these first avoids that situation.
+  await Promise.all([fetchJurisdictionByName(), fetchBloomJurisdictionsByName()])
```

Comment:
Thanks for documenting! I'm curious what the correct solution is to prevent this hack. I'm still new to ts and I'm not sure why this is preventing the functions from being called again- is it because they will be resolved promises after line 83?

Also, I don't understand the race condition. How will the results change if one of the `fetchJurisdictionByName` or `fetchBloomJurisdictionsByName` finishes before the other?

Reply:
(TODO for me  to update the comment slightly here to be more accurate, will not merge before then)

The issue is actually that both `fetchOpenListings` and `fetchClosedListings` call `fetchBloomJurisdictionsByName` at the same time which is a bad. Because of this line: https://github.com/metrotranscom/doorway/pull/29/files/26dab3f8a090859eafd64f810db8f0662f214995#diff-ebaa43fe70ada95e50f9e73f02838fb2611ce03f8e1caa91e9dbeee51e3ddf00R167.

The function basically checks to see if the arr is null, then does an http request if it is, then appends the result to the arr. However since is it called at the same time twice, it is null in both cases, then TWO http requests happen, then the jurisdictions get added twice. What happens in practice is that the `arr.length` is 2x the number actual jurisdictions if I don't do this (since it gets appended to twice, once when each promise resolves).

I honestly don't know what happens in the background to keep that jurisdictions arr around, but it does seem to stick around -- since the "caching" which is basically that jurisdictions arr instance variable somehow sticks around and `fetchBloomJurisdictionsByName` early returns.

Lastly I call `fetchJurisdictionByName` here at the same time for performance reasons and not race conditions reasons-- we might as well do those 2 concurrently and then do the /listings ones concurrently after. hence I should update the comment with all of this context

Reply:
I see. Thanks for the explaination! 

Now I'm wondering if we have an init stage of the application, where it can retrieve external data before coming online to serve requests.